transformer/transformer.go

```diff
@@ -0,0 +1,289 @@
+package transformer
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const startHighlightTag string = "<strong>"
+const endHighlightTag string = "</strong>"
+
+// Transformer represents an instance of the ResposeTransformer interface
+type Transformer struct {
+}
+
+type searchResponse struct {
+	Count        int           `json:"count"`
+	Took         int           `json:"took"`
+	Items        []contentItem `json:"items"`
+	ContentTypes []contentType `json:"content_types"`
+}
+type contentItem struct {
+	Description description `json:"description"`
+	Type        string      `json:"type"`
+	URI         string      `json:"uri"`
+	Matches     matches     `json:"matches,omitempty"`
+}
+type description struct {
+	Summary           string   `json:"summary"`
+	NextRelease       string   `json:"next_release,omitempty"`
+	Unit              string   `json:"unit,omitempty"`
+	Keywords          []string `json:"keywords,omitempty"`
+	ReleaseDate       string   `json:"release_date,omitempty"`
+	Edition           string   `json:"edition,omitempty"`
+	LatestRelease     bool     `json:"latest_release,omitempty"`
```

Comment:
All fields of boolean type may need to be a pointer, otherwise field will be set to the default value (false, not nil; to which the tag omitempty acts on and hence will always respond with a value even if there is not one set in elastic search document)

Reply:
Code updated to make optional fields pointers